FreeRTOS/Demo/Posix_GCC/CMakeLists.txt

```diff
@@ -0,0 +1,118 @@
+cmake_minimum_required(VERSION 3.15)
+
+project( posix_demo )
+
+set( CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -D_WINDOWS_")
+
+if( TRACE_ON_ENTER )
+    set( CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -DTRACE_ON_ENTER=1" )
+else()
+    set( CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -DTRACE_ON_ENTER=0" )
+endif()
+
+if( COVERAGE_TEST )
+    set( COVERAGE_TEST 1 )
+else()
+    set( COVERAGE_TEST 0 )
+endif()
+
+if( PROFILE )
+    set( CMAKE_BUILD_TYPE "release" )
+else()
+    set( CMAKE_BUILD_TYPE "debug" )
+endif()
+
+if( SANITIZE_ADDRESS )
+    set( CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fsanitize=address -fsanitize=alignment" )
+endif()
+
+if( SANITIZE_LEAK )
+    set( CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fsanitize=leak" )
+endif()
+
+set( FREERTOS_KERNEL_PATH "../../Source" )
+set( FREERTOS_PLUS_TRACE_PATH "../../../FreeRTOS-Plus/Source/FreeRTOS-Plus-Trace" )
+
+# Add the freertos_config for FreeRTOS-Kernel
+add_library( freertos_config INTERFACE )
+
+target_include_directories( freertos_config
+    INTERFACE
+        ./
+        ./Trace_Recorder_Configuration
+        ${FREERTOS_PLUS_TRACE_PATH}/Include
+)
+
+target_compile_definitions( freertos_config
+    INTERFACE
+        $<IF:${COVERAGE_TEST},projCOVERAGE_TEST=1,projCOVERAGE_TEST=0>
```

Comment:
This if statement seems redundant to me?
The check is for COVERAGE_TEST is already performed [here](https://github.com/FreeRTOS/FreeRTOS/pull/1020/files/7453f3942b5aff2d73468ef07a92c6b6c02a9842#diff-f43699e1da464edf8deec1a50cebdef2c647e14185fb39a36d17af179530f12eR13) on line 13. Why not set the `projCOVERAGE_TEST` variable in that if statement and then remove the check here?

Reply:
I think the original implementation has better consistency since COVERAGE_TEST is still required in other targets.

Variable needs to be defined to be used in cmake generator expression.
We can replace the definition with the following.
```
if( COVERAGE_TEST )
    set( COVERAGE_TEST 1 )
    set( CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -DprojCOVERAGE_TEST=1" )
else()
    set( COVERAGE_TEST 0 )
    set( CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -DprojCOVERAGE_TEST=0" )
endif()
```

However, COVERAGE_TEST variable is still required in the following target.
```
target_compile_options( freertos_kernel
    PRIVATE
        # Trace macro cast pointer to int to store memory management event
        $<IF:${COVERAGE_TEST},,-Wno-pointer-to-int-cast>
...
add_executable( posix_demo
                code_coverage_additions.c
                console.c
                main.c
                main_blinky.c
                main_full.c
                run-time-stats-utils.c
                $<$<NOT:${COVERAGE_TEST}>:${FREERTOS_PLUS_TRACE_SOURCES}>
...
)
```